Lombiq.Hosting.Tenants.FeaturesGuard/Startup.cs

```diff
@@ -1,14 +1,44 @@
 using Lombiq.Hosting.Tenants.FeaturesGuard.Constants;
 using Lombiq.Hosting.Tenants.FeaturesGuard.Handlers;
+using Lombiq.Hosting.Tenants.FeaturesGuard.Models;
+using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
 using OrchardCore.Environment.Shell;
+using OrchardCore.Environment.Shell.Configuration;
 using OrchardCore.Modules;
 
 namespace Lombiq.Hosting.Tenants.FeaturesGuard;
 
 [Feature(FeatureNames.FeaturesGuard)]
 public class Startup : StartupBase
 {
-    public override void ConfigureServices(IServiceCollection services) =>
+    private readonly IShellConfiguration _shellConfiguration;
+    private readonly IConfiguration _configuration;
+
+    public Startup(IConfiguration configuration, IShellConfiguration shellConfiguration)
+    {
+        _configuration = configuration;
+        _shellConfiguration = shellConfiguration;
+    }
+
+    public override void ConfigureServices(IServiceCollection services)
+    {
         services.AddScoped<IFeatureEventHandler, FeaturesEventHandler>();
+
+        if (_configuration.IsAzureHosting())
+        {
+            services.Configure<ConditionallyEnabledFeaturesOptions>(options =>
+                _shellConfiguration
+                    .GetSection("Lombiq_Hosting_Tenants_FeaturesGuard:ConditionallyEnabledFeaturesOptions:ConditionallyEnabledFeatures")
+                    .Bind(options));
+        }
+
+        if (_configuration.IsUITesting())
+        {
+            services.Configure<ConditionallyEnabledFeaturesOptions>(options =>
+                _shellConfiguration
+                    .GetSection("Lombiq_Hosting_Tenants_FeaturesGuard:TestConditionallyEnabledFeaturesOptions:ConditionallyEnabledFeatures")
+                    .Bind(options));
+        }
```

Comment:
These again tie the module to Azure (keep in mind that conditionally enabled features is not something just interesting for Azure), and the UI Testing Toolbox as well. Neither of these is needed.

Instead, these `Configure()` calls (or rather, only the first one, no need for a separate testing) can be here without conditions. And in the `Program`/`Startup` of the web app, you can add the configuration conditionally, as opposed to it being statically in appsettings. E.g.:

```c#
builder.Services
    .AddSingleton(configuration)
    .AddOrchardCms(orchardCoreBuilder =>
    {
        // ...

        if (configuration.IsAzureHosting())
        {
            orchardCoreBuilder.ConfigureServices(services =>
                services.Configure<ConditionallyEnabledFeaturesOptions>(options =>
                    options.EnableFeatureIfOtherFeatureIsEnabled = new Dictionary<string, string>
                    {
                        ["OrchardCore.Media.Azure.Storage"] = "OrchardCore.Media",
                    }));
        }
    });
```

Since we'll repeatedly use this, you can also add an extension method similar to `AllowMiniProfilerOnAdmin()` in Orchard, so one doesn't need to write the whole ceremony, just a) set `EnableFeatureIfOtherFeatureIsEnabled` if you have something custom or b) set the Azure Storage config directly since that'll be a common use-case. (I realize that this brings something from Azure into the module, but it doesn't require it to statically reference anything from Azure and still keeps the option open for any other config, like for Elasticsarch).

After this, configs from both sources (code, appsettings) will be merged, so you don't need the latter (but we keep the option open to use it). And you can remove the unneeded project/NuGet references from this project.